### PR TITLE
Implemented velocity boundary conditions and periodic boundary conditions

### PR DESCRIPTION
**Pull request by matheuscfernandes**
Reverts latticeboltzmann/2d-lb#20

### CODE REVIEW

**Comment by matheuscfernandes** [comment]
this reverts the revert, so it is essentially the original pull request. @btweinstein sorry for the mess! Also, there are some commented out stuff around which we can clean later on once we have the full functionality of what we want.

**Comment by btweinstein** [comment]
Ok so this is the pull request I should look at, right?

File: LB_D2Q9/pipe.py [hunk 1]
```diff
@@ -250,6 +250,171 @@ def run(self, num_iterations):
             self.collide_particles() # Relax the nonequilibrium fields
 
 
+class Pipe_Flow_PeriodicBC(Pipe_Flow):
+
+    def __init__(self, **kwargs):
+        super(Pipe_Flow_PeriodicBC, self).__init__(**kwargs)
+
+    def move_bcs(self):
```
Comment (btweinstein) [inline LB_D2Q9/pipe.py:258]: A couple of thoughts here. You need to update the hydrodynamic variables in addition to imposing the boundary conditions on f. So, you need to specify rho and u at the boundaries in the init_hydro class, right?

**Comment by btweinstein** [comment]
Also, if you do not initialize the same velocity everywhere, there should be a shockwave as you were seeing. So, you need to change the initial boundary condition with this as well.

**Comment by matheuscfernandes** [comment]
Aha! That makes sense. :) I will close this pull request because whenever I accepted your pull request and commited the changes I had crazy conflicts. I dont want this to happen with this. So, I am closing this and will commit it again once this problem is fixed.

**Pull request closed without merging**